test/e2e_node/e2e_build.go

```diff
@@ -0,0 +1,130 @@
+)
+
+var k8sBinDir = flag.String("k8s-bin-dir", "", "Directory containing k8s kubelet and kube-apiserver binaries.")
+
+func BuildGo() {
+	glog.Infof("Building k8s binaries...")
+	k8sRoot, err := getK8sRootDir()
+	if err != nil {
+		glog.Fatalf("Failed to locate kubernetes root directory %v.", err)
+	}
+	out, err := exec.Command(filepath.Join(k8sRoot, "hack/build-go.sh")).CombinedOutput()
+	if err != nil {
+		glog.Fatalf("Failed to build go packages %v.  Output:\n%s", err, out)
+	}
+}
+
+func getK8sBin(bin string) (string, error) {
+	// Use commandline specified path
+	if *k8sBinDir != "" {
+		absPath, err := filepath.Abs(*k8sBinDir)
+		if err != nil {
+			return "", err
+		}
+		if _, err := os.Stat(filepath.Join(*k8sBinDir, bin)); os.IsNotExist(err) {
+			return "", fmt.Errorf("Could not find kube-apiserver under directory %s.", absPath)
+		}
+		return filepath.Join(absPath, bin), nil
+	}
+
+	path, err := filepath.Abs(filepath.Dir(os.Args[0]))
+	if err != nil {
+		return "", fmt.Errorf("Could not find absolute path of directory containing the tests %s.", filepath.Dir(os.Args[0]))
+	}
+	if _, err := os.Stat(filepath.Join(path, bin)); !os.IsNotExist(err) {
+		return filepath.Join(path, bin), nil
+	}
+
+	buildOutputDir, err := getK8sBuildOutputDir()
+	if err != nil {
+		return "", err
+	}
+	if _, err := os.Stat(filepath.Join(buildOutputDir, bin)); !os.IsNotExist(err) {
+		return filepath.Join(buildOutputDir, bin), nil
+	}
+
+	// Give up with error
+	return "", fmt.Errorf("Unable to locate %s.  Can be defined using --k8s-path.", bin)
+}
+
+func getK8sRootDir() (string, error) {
+	// Get the directory of the current executable
+	_, testExec, _, _ := runtime.Caller(0)
+	path := filepath.Dir(testExec)
+
+	// Look for the kubernetes source root directory
+	if strings.Contains(path, "k8s.io/kubernetes") {
+		splitPath := strings.Split(path, "k8s.io/kubernetes")
+		return filepath.Join(splitPath[0], "k8s.io/kubernetes/"), nil
+	}
+
+	return "", fmt.Errorf("Could not find kubernetes source root directory.")
+}
+
+func getK8sBuildOutputDir() (string, error) {
+	k8sRoot, err := getK8sRootDir()
+	if err != nil {
+		return "", err
+	}
+	buildOutputDir := filepath.Join(k8sRoot, "_output/local/go/bin")
+	if _, err := os.Stat(buildOutputDir); os.IsNotExist(err) {
+		return "", err
+	}
+	return buildOutputDir, nil
+}
+
+func getK8sNodeTestDir() (string, error) {
+	k8sRoot, err := getK8sRootDir()
+	if err != nil {
+		return "", err
+	}
+	buildOutputDir := filepath.Join(k8sRoot, "test/e2e_node")
+	if _, err := os.Stat(buildOutputDir); os.IsNotExist(err) {
+		return "", err
+	}
+	return buildOutputDir, nil
+}
+
+func getKubeletServerBin() string {
+	bin, err := getK8sBin("kubelet")
+	if err != nil {
+		panic(fmt.Sprintf("Could not locate kubelet binary."))
+	}
+	return bin
+}
+
+func getApiServerBin() string {
+	bin, err := getK8sBin("kube-apiserver")
+	if err != nil {
+		panic(fmt.Sprintf("Could not locate kube-apiserver binary."))
+	}
+	return bin
+}
```

Comment:
I'm surprised that these functions don't already exist in test/e2e (non-node). Is there any common code you can reuse between the two suites? If it's a big refactoring it doesn't need to be in this PR.